### ISSUE TITLE
Better feedback on NoSuchElementException by adding the locator

### ISSUE THREAD

**Issue by SirLenz0rlot** [SeleniumHQ/selenium#1104]
Suggestion: put something like a `By#toString()` in the exception message.

I've implemented the PageObjects pattern using the PageFactory and I'm pretty happy with how things are setup right now, but one thing.
Once an element can't be found (e.g. because the selector has changed in the implementation), a `NoSuchElementException` is being thrown, that shows a lot of information on the current state (driver info, duration, version, session etc), except for the element that was being looked up, and I have to look quite deep in the stacktrace to find out at what point the automation fails. I notice that especially people who didn't write the automation have much trouble finding out why it fails.

Any reason not to do this?

An example stack trace of the current situation:
```
org.openqa.selenium.NoSuchElementException: no such element
  (Session info: chrome=47.0.2508.0)
  (Driver info: chromedriver=2.12.301324 (de8ab311bc9374d0ade71f7c167bad61848c7c48),platform=Linux 3.13.0-65-generic x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 45.04 seconds
For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
Build info: version: '2.44.0', revision: '76d78cf', time: '2014-10-23 20:03:00'
System info: host: 'mypc', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-65-generic', java.version: '1.7.0'
Session ID: c0ce3edee7e35f35b00ff07ef5923736
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{platform=LINUX, acceptSslCerts=true, javascriptEnabled=true, browserName=chrome, chrome={userDataDir=/tmp/.com.google.Chrome.SLbXex}, rotatable=false, locationContextEnabled=true, mobileEmulationEnabled=false, version=47.0.2508.0, takesHeapSnapshot=true, cssSelectorsEnabled=true, databaseEnabled=false, handlesAlerts=true, browserConnectionEnabled=false, webStorageEnabled=true, nativeEvents=true, applicationCacheEnabled=false, takesScreenshot=true}]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:86)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:58)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:542)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:204)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:156)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:352)
	at org.openqa.selenium.remote.RemoteWebDriver.findElementByCssSelector(RemoteWebDriver.java:441)
	at org.openqa.selenium.By$ByCssSelector.findElement(By.java:426)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:344)
	at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:59)
	at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:37)
	at com.sun.proxy.$Proxy15.findElement(Unknown Source)
	at <reference to my own code>
```

**Comment by alb-i986** [comment]
This seems to have already been fixed in db79088b9479e955fd9784ee7c6c168c61604ad7

> When Selenium is unable to interact with an element, such as the
>  case when an element is missing or disabled, this change will output
>  exactly how Selenium is attempting to locate the element, such as
>  the XPath or id of the element. This greatly speeds up
>  troubleshooting issues, as the exception message clearly states what
>  element is broken/missing for common problems like an
>  ElementNotFoundException.

@SirLenz0rlot Please try upgrading to 2.45.0 (the first release including that change) or later.

**Comment by SirLenz0rlot** [comment]
Thanks for the clarification and sorry, should've looked for newer versions before filing this.
